### PR TITLE
Issue #13345: Enable examples tests for javadocPackageCheck

### PR DESCRIPTION
**Pull request by AmitKumarDeoghoria**
### Title: Enable examples tests for  javadocPackageCheck
**Related Issue: #13345** 

**Summary**
This pull request aims to enable example tests for the **javadocPackageCheck** in the project. The implementation focuses on ensuring that   javadocPackageCheck are correctly validated through comprehensive test cases, improving overall code reliability and quality.

**Key Changes Made**

- Conversion of txt file to Java File
- Enable the test cases by adding expected violations in the expected array for all example tests
- Made corresponding changes in the xml and xml.template file
- Added package-info and package.html files for the check

### CODE REVIEW

**Comment by AmitKumarDeoghoria** [comment]
GitHub, generate website

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocpackage/legacy/Example2.java [hunk 1]
```diff
@@ -0,0 +1,14 @@
+/*xml
+<module name="Checker">
+  <module name="JavadocPackage">
+    <property name="allowLegacy" value="true"/>
+  </module>
+</module>
+*/
+// xdoc section -- start
+package com.puppycrawl.tools.checkstyle.checks.javadoc.javadocpackage.legacy;
+// OK as package.html file is present in directory
```
Comment (romani) [inline src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocpackage/legacy/Example2.java:10]: I hope you see yourself that generated documentation is weird as it reference to something that use does not see.
We need to print content of folder:
https://github.com/checkstyle/checkstyle/tree/a09d8d8cfa81b6d5a3115fd75ed85c866ac04514/src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocpackage/legacy

lets put in template file one more code block that show structure of folder:
```
Example2.java
package.html
``` 

please do for all examples.
Content of java files does not really matter, it is only useful to show that violation is placed on java files, even something is missed in folder.

So each example will he have 3 code blocks:
1) config macros
2) folder structure (hardcoded in as plain tags in xdoc template file)
3) java file content macros

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocpackage/legacywithboth/Example3.java [hunk 2]
```diff
@@ -0,0 +1,14 @@
+/*xml
+<module name="Checker">
+  <module name="JavadocPackage">
+    <property name="allowLegacy" value="true"/>
+  </module>
+</module>
+*/
+// xdoc section -- start
+package com.puppycrawl.tools.checkstyle.checks.javadoc.javadocpackage.legacywithboth;
+// violation 9 lines above 'Legacy package.html file should be removed'
```
Comment (romani) [inline src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocpackage/legacywithboth/Example3.java:10]: in html, line `9 lines above` looks weird, especially as user see "package ..." it is usual first line in java file.

lets do a trick: 
```
// xdoc section -- start
package com.puppycrawl.tools.checkstyle.checks.javadoc.javadocpackage.legacywithboth;
/* violation on first line 'Legacy package.html file should be removed' */
// xdoc section -- end` right after "package"
// violation 11 lines above 'Legacy package.html file should be removed'
.....
```

so we remove violation line from section that we print in html and keep multiline comment that is not recognized as special comment, so it is plain text, we can write anything we want.

**Review by romani — CHANGES_REQUESTED** [review]
items to improve:

**Comment by AmitKumarDeoghoria** [comment]
GitHub, generate website

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocpackage/legacy/Example2.java (hunk 1 above)
Comment (AmitKumarDeoghoria) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocpackage/legacy/Example2.java:10]: Done

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocpackage/legacywithboth/Example3.java (hunk 2 above)
Comment (AmitKumarDeoghoria) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocpackage/legacywithboth/Example3.java]: Done

File: src/xdocs/checks/javadoc/javadocpackage.xml [hunk 3]
```diff
@@ -51,22 +51,101 @@
...
 &lt;module name=&quot;Checker&quot;&gt;
   &lt;module name=&quot;JavadocPackage&quot;/&gt;
 &lt;/module&gt;
         </source>
+        <p id="Example1-structure">
+          <div class="wrapper">
+            <pre>
+src/xdocs-examples
+/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocpackage/nonlegacy/
+  Example1.java
```
Comment (romani) [inline src/xdocs/checks/javadoc/javadocpackage.xml:66]: just `Example1.java`, full path is not required, html site readers dont know it.

File: src/xdocs/checks/javadoc/javadocpackage.xml [hunk 4]
```diff
@@ -51,22 +51,101 @@
 
       <subsection name="Examples" id="Examples">
         <p id="Example1-config">
-          To configure the check:
+          To configure the check with no package-info.java file in directory:
```
Comment (romani) [inline src/xdocs/checks/javadoc/javadocpackage.xml:54]: put it in `  <p id="Example1-structure">`

**Review by romani — CHANGES_REQUESTED** [review]
items:

**Comment by AmitKumarDeoghoria** [comment]
GitHub, generate website

File: src/xdocs/checks/javadoc/javadocpackage.xml (hunk 3 above)
Comment (AmitKumarDeoghoria) [reply src/xdocs/checks/javadoc/javadocpackage.xml]: Done

File: src/xdocs/checks/javadoc/javadocpackage.xml (hunk 4 above)
Comment (AmitKumarDeoghoria) [reply src/xdocs/checks/javadoc/javadocpackage.xml]: Done

**Comment by AmitKumarDeoghoria** [comment]
GitHub, generate website

File: src/xdocs/checks/javadoc/javadocpackage.xml [hunk 5]
```diff
@@ -58,15 +58,77 @@
   &lt;module name=&quot;JavadocPackage&quot;/&gt;
 &lt;/module&gt;
         </source>
+        <p id="Example1-structure">
+          File structure with no package-info.java file in directory:<br/><br/>
```
Comment (romani) [inline src/xdocs/checks/javadoc/javadocpackage.xml:62]: Directory structure .....


Fix in all cases

**Review by romani — CHANGES_REQUESTED** [review]
Items

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocpackage/legacy/Example2.java [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+public class Example2 {
+  String str="some content";
+}
```
Comment (romani) [inline src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocpackage/legacy/Example2.java:13]: `public class Example2 { }`

For all, to let people easily see that class content doesn't matter

**Review by romani — CHANGES_REQUESTED** [review]
Items  for all Examples

File: src/xdocs/checks/javadoc/javadocpackage.xml (hunk 5 above)
Comment (AmitKumarDeoghoria) [reply src/xdocs/checks/javadoc/javadocpackage.xml]: Done

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocpackage/legacy/Example2.java (hunk 6 above)
Comment (AmitKumarDeoghoria) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocpackage/legacy/Example2.java]: Done

**Comment by AmitKumarDeoghoria** [comment]
GitHub, generate website

**Review by romani — APPROVED** [review]
Awesome results

**Pull request merged**